commands/import.go

```diff
@@ -102,40 +106,51 @@ func checkFilesExist(files []string) error {
 	return nil
 }
 
-// setRolling determines what the current rolling and chunk settings should be
-// based on the values already in the database and the values supplied on the
-// command line by the user.
-func setRolling(dbExists bool, dbIsRolling bool, dbCurrChunk int, dbTotalChunks int,
-	userIsRolling bool, userCurrChunk int, userTotalChunks int, cfgDefaultChunks int) (config.RollingStaticCfg, error) {
+// parseFlags validates the user supplied flags against the current state of
+// the target database and determines what settings should be set in the
+// rolling configuration
+func parseFlags(dbExists bool, dbIsRolling bool, dbCurrChunk int, dbTotalChunks int,
+	userIsRolling bool, userCurrChunk int, userTotalChunks int, cfgDefaultChunks int,
+	deleteOldData bool) (config.RollingStaticCfg, error) {
 
 	cfg := config.RollingStaticCfg{}
 
-	// Can uncomment this check if we want to preserve old behavior with regards to non-rolling databases
-	// if dbExists && !dbIsRolling {
-	// 	return cfg, errors.New("\n\t[!] New data cannot be imported into an existing non-rolling database")
-	// }
-
 	// a user-provided value for either of the chunk options implies rolling
 	if userTotalChunks != -1 || userCurrChunk != -1 {
 		userIsRolling = true
 	}
 
+	// ensure the user specifies a rolling import if the database exists
+	// and is not a rolling database.
+	if !deleteOldData && (dbExists && !dbIsRolling) && !userIsRolling {
```

Comment:
Force the user to specify the rolling flag or a chunk flag to convert an existing non-rolling dataset to a rolling dataset.